daemon/api.go

```diff
@@ -535,12 +525,7 @@ var getConfigurator = func() configurator {
 
 func snapConfig(c *Command, r *http.Request) Response {
 	vars := muxVars(r)
-	name := vars["name"]
-	developer := vars["developer"]
-	if name == "" || developer == "" {
-		return BadRequest("missing name or developer")
-	}
-	pkgName := name + "." + developer
+	snapName := vars["name"]
```

Comment:
@stevenwilkin because name comes from the pattern /{name}/ in the path which I think means it now cannot be empty